parquet/metadata/page_index.go

```diff
@@ -0,0 +1,889 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"fmt"
+	"io"
+	"math"
+
+	"github.com/JohnCGriffin/overflow"
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/parquet"
+	"github.com/apache/arrow-go/v18/parquet/internal/debug"
+	"github.com/apache/arrow-go/v18/parquet/internal/encoding"
+	"github.com/apache/arrow-go/v18/parquet/internal/encryption"
+	format "github.com/apache/arrow-go/v18/parquet/internal/gen-go/parquet"
+	"github.com/apache/arrow-go/v18/parquet/internal/thrift"
+	"github.com/apache/arrow-go/v18/parquet/internal/utils"
+	"github.com/apache/arrow-go/v18/parquet/schema"
+)
+
+type BoundaryOrder = format.BoundaryOrder
+
+const (
+	Unordered  BoundaryOrder = format.BoundaryOrder_UNORDERED
+	Ascending  BoundaryOrder = format.BoundaryOrder_ASCENDING
+	Descending BoundaryOrder = format.BoundaryOrder_DESCENDING
+)
+
+type ColumnIndex interface {
```

Comment:
It might be useful to have docstrings for public types at least?

Reply:
bah! I thought I had gone through and added the docstrings, I guess I missed a few. I'll fix that!